### DISCUSSION TITLE
Compilation issues, problems running code?

### DISCUSSION THREAD

**Discussion by tomlankhorst** [leggedrobotics/darknet_ros#282] [General]
Feel free to discuss the issues that you encounter on the new Discussions page!

**Comment by yifanshi86** [comment]
I encountered the following error "[ 98%] Generating C++ code from darknet_ros_msgs/CheckForObjectsResult.msg
[ 98%] Built target darknet_ros_msgs_generate_messages_cpp
[ 98%] Built target graspit_interface_gencpp
[ 98%] Built target graspit_interface_generate_messages
Scanning dependencies of target darknet_ros_lib
make[2]: *** No rule to make target 'darknet_ros/darknet_ros/CMakeFiles/darknet_ros_lib.dir/build'.  Stop.
CMakeFiles/Makefile2:5241: recipe for target 'darknet_ros/darknet_ros/CMakeFiles/darknet_ros_lib.dir/all' failed
make[1]: *** [darknet_ros/darknet_ros/CMakeFiles/darknet_ros_lib.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 98%] Built target darknet_ros_msgs_generate_messages_eus
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
Invoking "make -j2 -l2" failed
" 
with ubuntu 18.04 and melodic ROS. openCV version is 3.2.0

**Comment by yifanshi86** [comment]
I started a brand new catkin workspace and put melodic version of darknet_ros into the src folder, and did catkin_make. the failed message is "CMake Error: cannot find source fiel of activation_layer.c".

"-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib/x86_64-linux-gnu/libX11.so
--  X11_INCLUDE_DIR: /usr/include
--  X11_LIBRARIES: /usr/lib/x86_64-linux-gnu/libSM.so/usr/lib/x86_64-linux-gnu/libICE.so/usr/lib/x86_64-linux-gnu/libX11.so/usr/lib/x86_64-linux-gnu/libXext.so
-- Boost  found.
-- Found Boost components:
   thread
-- Found OpenCV: /usr/local (found version "4.5.2") 
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
-- Checking and downloading yolov2-tiny.weights if needed ...
-- ... file does not exist. Downloading now ...
-- Checking and downloading yolov3.weights if needed ...
-- Checking and downloading yolov2.weights if needed ...
-- Configuring done
CMake Error at darknet_ros/darknet_ros/CMakeLists.txt:169 (add_library):
  Cannot find source file:

    DARKNET_PATH-NOTFOUND/src/activation_layer.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx


CMake Error: CMake can not determine linker language for target: darknet_ros_lib
CMake Error: Cannot determine link language for target "darknet_ros_lib".
-- Generating done
-- Build files have been written to: /home/yifan/catkin_ws/build
Invoking "cmake" failed
"

**Reply by tomlankhorst** [reply]
Looks like you haven't cloned recursively since darknet is not found.

**Comment by yifanshi86** [comment]
Thanks, that is helpful. I recursively cloned the source code, and the error message is gone.
I had another issue running the launch file. When I run "roslaunch darknet_ros darknet_ros.launch", it has the following failed message:
"... logging to /home/yifan/.ros/log/57075624-9f45-11eb-9d88-0800277c1c0e/roslaunch-yifan-VirtualBox-14309.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://yifan-VirtualBox:32779/

SUMMARY
========

PARAMETERS
 * /darknet_ros/actions/camera_reading/name: /darknet_ros/chec...
 * /darknet_ros/config_path: /home/yifan/catki...
 * /darknet_ros/image_view/enable_console_output: True
 * /darknet_ros/image_view/enable_opencv: True
 * /darknet_ros/image_view/wait_key_delay: 1
 * /darknet_ros/publishers/bounding_boxes/latch: False
 * /darknet_ros/publishers/bounding_boxes/queue_size: 1
 * /darknet_ros/publishers/bounding_boxes/topic: /darknet_ros/boun...
 * /darknet_ros/publishers/detection_image/latch: True
 * /darknet_ros/publishers/detection_image/queue_size: 1
 * /darknet_ros/publishers/detection_image/topic: /darknet_ros/dete...
 * /darknet_ros/publishers/object_detector/latch: False
 * /darknet_ros/publishers/object_detector/queue_size: 1
 * /darknet_ros/publishers/object_detector/topic: /darknet_ros/foun...
 * /darknet_ros/subscribers/camera_reading/queue_size: 1
 * /darknet_ros/subscribers/camera_reading/topic: /camera/rgb/image...
 * /darknet_ros/weights_path: /home/yifan/catki...
 * /darknet_ros/yolo_model/config_file/name: yolov2-tiny.cfg
 * /darknet_ros/yolo_model/detection_classes/names: ['person', 'bicyc...
 * /darknet_ros/yolo_model/threshold/value: 0.3
 * /darknet_ros/yolo_model/weight_file/name: yolov2-tiny.weights
 * /rosdistro: melodic
 * /rosversion: 1.14.10

NODES
  /
    darknet_ros (darknet_ros/darknet_ros)

ROS_MASTER_URI=http://localhost:11311

process[darknet_ros-1]: started with pid [14324]
[ INFO] [1618640672.418408217]: [YoloObjectDetector] Node started.
[ INFO] [1618640672.425180178]: [YoloObjectDetector] Xserver is running.
[ INFO] [1618640672.426771847]: [YoloObjectDetector] init().
YOLO
First section must be [net] or [network]: Resource temporarily unavailable
[darknet_ros-1] process has died [pid 14324, exit code -11, cmd /home/yifan/catkin_ws/devel/lib/darknet_ros/darknet_ros camera/rgb/image_raw:=/camera/rgb/image_raw __name:=darknet_ros __log:=/home/yifan/.ros/log/57075624-9f45-11eb-9d88-0800277c1c0e/darknet_ros-1.log].
log file: /home/yifan/.ros/log/57075624-9f45-11eb-9d88-0800277c1c0e/darknet_ros-1*.log
all processes on machine have died, roslaunch will exit
shutting down processing monitor...
... shutting down processing monitor complete
done
"
Is that because camera is not available? I used ubuntu 18 virtual machine.

**Comment by ahengtell** [comment]
"roslaunch darknet_ros darknet_ros.launch roslaunch usb_cam usb_cam-test.launch"when I start darknet_ros node while cam_node is running, the darknet_ros node shows the process has died.the detail of error message is below:
Waiting for image.
Waiting for image.
[darknet_ros-2] process has died [pid 10262, exit code -11, cmd /home/aheng/桌面/ROBOTICWS/devel/lib/darknet_ros/darknet_ros camera/rgb/image_raw:=/camera/rgb/image_raw __name:=darknet_ros __log:=/home/aheng/.ros/log/d3441234-3fc8-11ed-9706-336173c45f0f/darknet_ros-2.log].
log file: /home/aheng/.ros/log/d3441234-3fc8-11ed-9706-336173c45f0f/darknet_ros-2*.log
^C[rosout-1] killing on exit
[master] killing on exit
shutting down processing monitor...
... shutting down processing monitor complete
done
